Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3560 @@
+
+        else:
+            number_of_requests = 1
+            offset = None
+            offset_multiplier = 1
+
+    # Manual Pagination
+    elif is_manual:
+        page = page or 1
+        page_size = page_size or 1
+        number_of_requests = 1
+        limit = page_size
+        offset = (page - 1) * page_size
+        offset_multiplier = 1
+
+    # No Pagination
+    else:
+        number_of_requests = 1
+        limit = MAX_PAGE_SIZE
+        offset = None
+        offset_multiplier = 1
+
+    return Pagination(page, page_size, limit, offset, number_of_requests, offset_multiplier, is_automatic, is_manual)
+
+
+def extract_pagination_from_response(pagination: Pagination, raw_response: Dict[str, Any]) -> Tuple[List, str]:
+    """
+    Extract values from the response according to pagination parameters.
+
+    Args:
+        pagination (Pagination): Pagination parameters to extract values according to.
+        raw_response (Dict[str, Any]): Raw response to extract values from.
+
+    Returns:
+        Tuple[List, str]: Context output and Readable output.
+    """
+    if pagination.is_manual:
+        start = (pagination.page - 1) * pagination.page_size
+        stop = pagination.page * pagination.page_size
+
+        raw_response['data']['events'] = raw_response['data']['events'][start:stop]
+
+    else:
+        raw_response['data']['events'] = raw_response['data']['events'][:pagination.limit]
+
+    context_output = get_context_output(raw_response, ['links'])
+    context_output = context_output[0]['events']
+    add_item_to_all_dictionaries(
+        context_output,
+        'connector_guid',
+        dict_safe_get(raw_response, ['data', 'computer', 'connector_guid'])
+    )
+
+    readable_output = get_computer_readable_output(raw_response)
+    readable_output += get_readable_output(
+        response=raw_response,
+        header_by_keys=TRAJECTORY_HEADERS_BY_KEYS,
+        keys_to_items_option_1=['data', 'events'],
+        title=TRAJECTORY_TITLE,
+    )
+
+    return context_output, readable_output
+
+
+def delete_keys_from_dict(dictionary: MutableMapping, keys_to_delete: List[str] | Set[str]) -> Dict[str, Any]:
+    """
+    Get a modified dictionary without the requested keys
+
+    Args:
+        dictionary (Dict[str, Any]): Dictionary to modify according to.
+        keys_to_delete (List[str]): Keys to not include in the modified dictionary.
+
+    Returns:
+        Dict[str, Any]: Modified dictionary without requested keys.
+    """
+    keys_set = set(keys_to_delete)
+    modified_dict: Dict[str, Any] = {}
+
+    for key, value in dictionary.items():
+        if key not in keys_set:
+            if isinstance(value, MutableMapping):
+                modified_dict[key] = delete_keys_from_dict(value, keys_set)
+
+            elif isinstance(value, MutableSequence) \
+                    and len(value) > 0 \
+                    and isinstance(value[0], MutableMapping):
+                modified_dict[key] = []
```

Comment:
why `[]` ?

Reply:
My mistake, changed that line and the loop below to a list comprehension.